js/main.js

```diff
@@ -20,23 +23,51 @@ function logForm(event) {
     notes: $textArea.value,
     entryId: data.nextEntryId
   };
-  data.nextEntryId += 1;
-  data.entries.unshift(formObject);
-  $setImg.setAttribute('src', 'images/placeholder-image-square.jpg');
-  $formSubmit.reset();
-  $ul.prepend(entryDOM(formObject));
-  switchViews("entries");
+  if (data.editing !== null){
+    for (var k = 0; k < data.entries.length; k++) {
+      if (data.editing === data.entries[k].entryId) {
+        var changeObject = {
+          title: $inputTitle.value,
+          photoUrl: $imgUrl.value,
+          notes: $textArea.value,
+          entryId: data.entries.length - k
+        }
+        data.entries.splice(k, 1, changeObject)
+        var $findLi = document.querySelectorAll('li')
+        $findLi[k].replaceWith(entryDOM(changeObject))
+        data.editing = null;
+      }
+    }
+  } else {
+      data.nextEntryId += 1;
+      data.entries.unshift(formObject);
+      $setImg.setAttribute('src', 'images/placeholder-image-square.jpg');
+      $formSubmit.reset();
+      $ul.prepend(entryDOM(formObject));
+    }
+   switchViews("entries");
+   
 }
 
 function entryDOM(entry) {
+    var $icon = document.querySelectorAll('i');
+    for (var i = 0; i < $icon.length; i++) {
+      $icon[i].setAttribute("EntryId", $icon.length - i)
+      $icon[i].addEventListener('click', function () {
+        switchViews(event.target.getAttribute('data-view'))
+      })
+    }
```

Comment:
Instead of doing this loop every time a new DOM tree is created, you should either apply the event listener to the edit icon element right after you create it, or use event delegation and apply the event handler to the `ul` holding all the entries.